pkg/bmc/redfish.go

```diff
@@ -30,11 +31,22 @@ func newRedfishVirtualMediaAccessDetails(parsedURL *url.URL) (AccessDetails, err
 	}, nil
 }
 
+func newiDracRedfishVirtualMediaAccessDetails(parsedURL *url.URL) (AccessDetails, error) {
+	return &redfishAccessDetails{
+		bmcType:        parsedURL.Scheme,
+		host:           parsedURL.Host,
+		path:           parsedURL.Path,
+		isVirtualMedia: true,
+		isiDrac:        true,
+	}, nil
+}
+
 type redfishAccessDetails struct {
 	bmcType        string
 	host           string
 	path           string
 	isVirtualMedia bool
+	isiDrac        bool
```

Comment:
There are a couple of alternative ways of implementing the pattern

    if a.isiDrac {
        ...
    }

that could be worth considering. One would be to have an enumeration for the vendor, and then use e.g.

    switch a.vendor {
        case vendor_iDrac:
            ...
    }

and another is to create a separate type that can nevertheless 'inherit' (sort-of) methods that are the same (like `DriverInfo()`) while overriding others:

    type iDracRedfishAccessDetails struct {
        redfishAccessDetails
    }
    func (a *iDracRedfishAccessDetails) Driver() string {
        return "idrac"
    }
    ....

Reply:
I like the [embedded interface](https://golang.org/doc/effective_go.html#embedding) to override functions.

isVirtualMedia and then isiDrac (which is also virtual media) is a little bit confusing to me

Reply:
`is*` fields removed, unnecessary conditionals removed, and all the new types added :smile: